integration-cli/docker_cli_build_test.go

```diff
@@ -4632,10 +4632,10 @@ func (s *DockerCLIBuildSuite) TestBuildMultiStageArg(c *testing.T) {
 	parentID := imgs[0]
 
 	result = cli.DockerCmd(c, "run", "--rm", parentID, "cat", "/out")
-	assert.Assert(c, strings.Contains(result.Stdout(), "foo=abc"))
+	assert.Assert(c, is.Contains(result.Stdout(), "foo=abc"))
 	result = cli.DockerCmd(c, "run", "--rm", imgName, "cat", "/out")
 	assert.Assert(c, !strings.Contains(result.Stdout(), "foo"))
```

Comment:
Still looking for more elegant solutions to these (negative matches); we could probably consider putting `result.Stdout()` as part of a custom failure message to help (definitely not needed for this PR though)


edit: ah; I see you mentioned this in your PR description as well

> Fix inverse comparisons. integration-cli/checker doesn't help with this entirely

Reply:
Yeah, the trick is that we need to create a custom type that wraps `assert.BoolOrComparison` which then inverts `Success` as well as providing a wrapped `FormatMessage` that describes that the comparison was inverted. I suspect the wrapped description is going to be quite janky (it would probably look like the suggestion that the author of `gotest.tools` proposed in https://github.com/gotestyourself/gotest.tools/issues/147#issuecomment-622200454). Unfortunately, `FormatMessage` is technically internal to `gotest.tools` so it could break in the future (but the author suggested doing it this way, so I'm not sure).

Since it seems you'd be happy with a patch like that, I can try to cook one up.

Reply:
Yeah, a custom helper could work still, but perhaps we could still look if we could contribute a patch to `gotest.tools` that would be acceptable to their maintainers 🤔 


For the formatting, we could still go the manual way;

```go
assert.Assert(c, !strings.Contains(result.Stdout(), "foo"), "Expected: %q to contain %q", result.Stdout(), "foo")
```

Which (if all of these are in the same (`integration-cli`) package could be a local helper I guess;

```go
func assertNotContains(t testing.T, val, expected string) {
    t.Helper()
    assert.Assert(t, !strings.Contains(val, expected), "Expected %q not to contain %q", val, expected)
}
```

Reply:
That would be the simplest option if we want to avoid over-complicating things. I'll go with that version now and we can do the more complicated version in the future...